src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl

```diff
@@ -107,24 +107,28 @@ then
 fi
 
 {{ if CompressNodeModulesCommand | IsNotBlank }}
-if [ "$SOURCE_DIR" != "$DESTINATION_DIR" ]
-then
-	if [ -f $zippedModulesFileName ]; then
-		echo
-		echo "File '$zippedModulesFileName' already exists under '$SOURCE_DIR'. Deleting it ..."
-		rm -f $zippedModulesFileName
-	fi
-
-	if [ -d node_modules ]
+postPostBuildScriptExecute="compressNodeModules"
+function compressNodeModules()
+{
+	if [ "$SOURCE_DIR" != "$DESTINATION_DIR" ]
 	then
-		echo
-		echo Zipping existing 'node_modules' folder ...
-		START_TIME=$SECONDS
-		# Make the contents of the node_modules folder appear in the zip file, not the folder itself
-		cd node_modules
-		{{ CompressNodeModulesCommand }} ../$zippedModulesFileName .
-		ELAPSED_TIME=$(($SECONDS - $START_TIME))
-		echo "Done in $ELAPSED_TIME sec(s)."
+		if [ -f $zippedModulesFileName ]; then
+			echo
+			echo "File '$zippedModulesFileName' already exists under '$SOURCE_DIR'. Deleting it ..."
+			rm -f $zippedModulesFileName
```

Comment:
I think that the zip tool optimizes for this case when the file is already there, only adding files.

Reply:
I am not doing this change in this PR any more and only exposing the source and destination directories.